dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -520,6 +521,184 @@ private void handleEntry2tsv(List<String> args) throws ApiException, IOException
     }
 
     /**
+     * this function will check if the content of the file is CWL or not
+     * @param content
+     * @return
+     */
+    private Boolean checkCWL(File content){
+        /* CWL: check for 'class:Workflow OR CommandLineTool', 'inputs: ','outputs: ', and 'steps' */
+        Pattern cwlVersion = Pattern.compile("(.*)(cwlVersion)(.*)(:)(.*)");
+        Pattern classWfPattern = Pattern.compile("(.*)(class)(.*)(:)(\\sWorkflow)");
+        Pattern classToolPattern = Pattern.compile("(.*)(class)(.*)(:)(\\sCommandLineTool)");
+        String line;
+        Boolean versionFound = false;
+
+        try{
+            InputStreamReader is = new InputStreamReader(new FileInputStream(content.getPath()),StandardCharsets.UTF_8);
```

Comment:
CWL files are reasonably small. Just use something simpler like the nio utility methods for reading a file https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#readAllLines-java.nio.file.Path-java.nio.charset.Charset-